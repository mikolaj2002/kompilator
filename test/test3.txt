# Test 3. Conditions

PROGRAM IS
    value_true, value_false, x, y
IN

    value_true := 100;
    value_false := 69;

    IF 2 < 3 THEN
        WRITE value_true;
    ELSE
        WRITE value_false;
    ENDIF

    x := 15;
    y := 16;
    IF y >= x THEN
        WRITE value_true;
        IF y = x THEN
            WRITE value_true;
        ELSE
            WRITE value_false;
        ENDIF
    ELSE
        WRITE value_false;
    ENDIF

    IF x != y THEN
        WRITE value_true;
    ELSE
        WRITE value_false;
    ENDIF

    IF x > y THEN
        WRITE value_true;
    ELSE
        IF y <= x THEN
            WRITE value_true;
        ELSE
            WRITE value_false;
        ENDIF
    ENDIF

END