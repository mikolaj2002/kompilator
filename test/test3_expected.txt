100
100
69
100
69
